packages/nextjs/hooks/scaffold-eth/useScaffoldContractWrite.tsx

```diff
@@ -34,9 +36,15 @@ export const useScaffoldContractWrite = (contractName: string, functionName: str
       return;
     }
 
+    // Looking at blockchain to see whats stored at `deployedContractData.address`, if its `0x0` then high possibility that its not a contract
+    const code = await provider.getCode(deployedContractData.address);
+    if (code === "0x") {
+      toast.error(<NoContractAddressError address={deployedContractData.address} />);
+      return;
+    }
+
```

Comment:
Check the ToDo comment I wrote last week here: https://github.com/scaffold-eth/se-2/blob/main/packages/nextjs/components/scaffold-eth/Contract/utilsContract.tsx#L36

I feel it makes more sense to have this logic inside `getDeployedContract`.

If we do that, we could have the same error message for two cases you mentioned... since are both the same (contract is not deployed.)

```ts
    if (!deployedContractData) {
      toast.error("....");
      return;
    }
```

Reply:
> I feel it makes more sense to have this logic inside getDeployedContract.

Yup agree 💯, replaced `getDeployedContract` util with [useDeployedContractInfo](https://github.com/scaffold-eth/se-2/pull/130/commits/ad8ee98a7ba5af6ca146e15255a3f736babd9c21) hook so that we levearage wagmi's usefull hooks internally like `useProvider` instead of everytime creating it and then passing it to `getDeployedContract` util. 


PS : feel free to change the hook name and also happy to consider any approach

Reply:
Love using a hook!